src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.props

```diff
@@ -14,10 +14,14 @@ The .NET Foundation licenses this file to you under the MIT license.
 -->
 <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
-    <CppCompilerAndLinker Condition="'$(CppCompilerAndLinker)' == ''">clang</CppCompilerAndLinker>
+    <CustomToolchainSpecified Condition="'$(CppCompilerAndLinker)' != ''">true</CustomToolchainSpecified>
+    <CppCompilerAndLinker Condition="'$(CustomToolchainSpecified)' != 'true'">clang</CppCompilerAndLinker>
+    <CppCompilerAndLinkerAlternative Condition="'$(CustomToolchainSpecified)' != 'true' and $(CppCompilerAndLinker.ToLowerInvariant().Contains('clang'))">gcc</CppCompilerAndLinkerAlternative>
```

Comment:
Should we ship setting `CppCompilerAndLinkerAlternative` on OSX so that everything related to gcc is skipped there?

Reply:
Certainly. Although it is possible to install gcc via brew on macOS. I think it requires XCode command line tools to be installed (which comes with AppleClang). If someone really want to compile native code via gcc on OSX, they can set `CppCompilerAndLinker` explicitly. This automation is only intended for linux distros, where it is common to have gcc present in PATH.